fhir-model/src/test/java/com/ibm/fhir/model/spec/test/R4ExamplesDriver.java

```diff
@@ -206,6 +253,68 @@ public void processIndex(String filename, Format format) throws Exception {
     public void processExample(String file, Expectation expectation) throws ExampleProcessorException {
         processExample(file, Format.JSON, expectation);
     }
+    
+    /**
+     * Submit the given example for processing
+     * @param errors the list of errors to accumulate
+     * @param file
+     * @param format
+     * @param expectation
+     * @throws ExampleProcessorException
+     */
+    public void submitExample(List<ExampleProcessorException> errors, String file, Format format, Expectation expectation) throws ExampleProcessorException {
+        if (pool != null) {
+            
+            // Wait until we have capacity. We do this to throttle the number of requests
+            // submitted to pool, hopefully avoiding memory issues if ever we have a really
+            // large index to process
+            lock.lock();
+            while (this.currentlySubmittedCount == this.maxInflight) {
+                try {
+                    this.inflightCondition.await(1000, TimeUnit.MILLISECONDS);
+                }
+                catch (InterruptedException x) {
+                    // NOP
+                }
+            }
+            currentlySubmittedCount++;
+            lock.unlock();
+            
+            pool.execute(new Runnable() {
+
+                @Override
+                public void run() {
+                    try {
+                        processExample(file, format, expectation);
+                    } catch (ExampleProcessorException e) {
+                        lock.lock();
+                        errors.add(e);
+                        lock.unlock();
+                    }
+                    finally {
+                        lock.lock();
+                        int oldCount = currentlySubmittedCount--;
+                        if (oldCount == maxInflight) {
+                            inflightCondition.signal();
+                        }
+                        
+                        if (currentlySubmittedCount == 0) {
+                            runningCondition.signal();
+                        }
+                        lock.unlock();
+                    }
+                }
+            });
+        }
+        else {
+            // run in-line
+            try {
+                processExample(file, format, expectation);
+            } catch (ExampleProcessorException e) {
+                errors.add(e);
+            }
+        }
+    }
```

Comment:
maybe this class should be abstract, and this code should go up a level to the persistence layer or the server-tests layer?